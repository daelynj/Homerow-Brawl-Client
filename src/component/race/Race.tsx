import * as React from "react";
import { Player } from "./Player";

interface Props {
  wordsLength: number;
  currentWordIndex: number;
  updatePosition: (newPosition: any) => void;
  raceState: any;
  ID: number;
}

export const Race = (props: Props) => {
  const generateRace = () =>
    props.raceState.players.map((player: any, index: number) => (
      <Player key={index} position={player.position} />
    ));

  const updatePosition = () => {
    let new_position: number =
      (props.currentWordIndex / props.wordsLength) * 100;
    let old_position: number = props.raceState.players.find(byID).position;

    if (old_position !== new_position) {
      props.updatePosition(new_position);
    }
  };

  const byID = (player: any) => {
    if (player.id === props.ID) {
      return player;
    }
  };

  const raceStart = () => {
    if (props.raceState !== null) {
      return true;
    }
  };

  return (
    <>
      {raceStart() && generateRace()}
      {raceStart() && updatePosition()}
    </>
  );
};
