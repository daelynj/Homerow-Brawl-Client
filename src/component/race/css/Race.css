.race {
  margin-left: auto;
  margin-right: auto;
  width: 62%;
  padding-right: 10%;
  background-color: white;
  border: 5px solid black;
}
