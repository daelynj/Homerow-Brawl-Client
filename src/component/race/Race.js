import React from "react";
import octane from "./images/octane.png";
import "./css/Race.css";

function Race(props) {
  const wordsToAdvance = Math.round(props.wordsLength / 15);

  function calculatePosition() {
    return "position" + Math.floor(props.currentWordIndex / wordsToAdvance);
  }
  return (
    <div className={calculatePosition()}>
      <img src={octane} alt="Octane" />
    </div>
  );
}

export default Race;
