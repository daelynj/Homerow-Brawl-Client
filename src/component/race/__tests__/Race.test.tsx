import * as React from "react";
import { Race } from "../Race";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const updatePosition = jest.fn();

describe("Race", () => {
  const buildProps = (newProps = {}) => ({
    wordsLength: 40,
    currentWordIndex: 0,
    updatePosition,
    raceState: { players: [{ id: 1, position: 0 }] },
    ID: 1,
    ...newProps
  });

  it("renders an image at position 0", () => {
    const RaceComponent = mount(<Race {...buildProps()} />);
    expect(RaceComponent.find("img").exists()).toBe(true);

    const tree = renderer.create(<Race {...buildProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders an image at position 10%", () => {
    const RaceComponent = mount(
      <Race {...buildProps({ currentWordIndex: 4 })} />
    );

    expect(RaceComponent.find("img").exists()).toBe(true);

    const tree = renderer
      .create(<Race {...buildProps({ currentWordIndex: 4 })} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders two images", () => {
    const payload = {
      players: [{ id: 1, position: 0 }, { id: 2, position: 0 }]
    };
    const RaceComponent = mount(
      <Race {...buildProps({ raceState: payload })} />
    );

    expect(RaceComponent.find("img").exists()).toBe(true);

    const tree = renderer
      .create(<Race {...buildProps({ raceState: payload })} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
