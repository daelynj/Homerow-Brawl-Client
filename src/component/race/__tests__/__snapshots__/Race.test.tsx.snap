// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Race renders an image at position 0 1`] = `
<div
  className="race"
>
  <div
    style={
      Object {
        "paddingLeft": "0%",
        "textAlign": "left",
      }
    }
  >
    <img
      alt="Octane"
      src="octane.png"
    />
    <div />
  </div>
</div>
`;

exports[`Race renders an image at position 10% 1`] = `
<div
  className="race"
>
  <div
    style={
      Object {
        "paddingLeft": "0%",
        "textAlign": "left",
      }
    }
  >
    <img
      alt="Octane"
      src="octane.png"
    />
    <div />
  </div>
</div>
`;

exports[`Race renders two images 1`] = `
<div
  className="race"
>
  <div
    style={
      Object {
        "paddingLeft": "0%",
        "textAlign": "left",
      }
    }
  >
    <img
      alt="Octane"
      src="octane.png"
    />
    <div />
  </div>
  <div
    style={
      Object {
        "paddingLeft": "0%",
        "textAlign": "left",
      }
    }
  >
    <img
      alt="Octane"
      src="octane.png"
    />
    <div />
  </div>
</div>
`;
